cosmic-app-list/src/wayland_handler.rs

```diff
@@ -135,12 +135,23 @@ impl ToplevelInfoHandler for AppData {
         toplevel: &zcosmic_toplevel_handle_v1::ZcosmicToplevelHandleV1,
     ) {
         if let Some(info) = self.toplevel_info_state.info(toplevel) {
-            let _ = self
-                .tx
-                .unbounded_send(WaylandUpdate::Toplevel(ToplevelUpdate::Add(
-                    toplevel.clone(),
-                    info.clone(),
-                )));
+            if !info
+                .state
+                .contains(&zcosmic_toplevel_handle_v1::State::Minimized)
+            {
+                let _ = self
+                    .tx
+                    .unbounded_send(WaylandUpdate::Toplevel(ToplevelUpdate::Add(
+                        toplevel.clone(),
+                        info.clone(),
+                    )));
+            } else {
+                let _ = self
+                    .tx
+                    .unbounded_send(WaylandUpdate::Toplevel(ToplevelUpdate::Remove(
+                        toplevel.clone(),
+                    )));
+            }
```

Comment:
I wonder if this might be problematic in certain situations.

E.g. if you don't have the minimize-applet, you can't use the dock to activate the windows again, which people with a more traditional single-panel layout might want.

Additionally you cannot select context options like Firefox "New Private Window", if your window is minimized, because it will always launch a new instance, which is counter-intuitive. (And might even be promise wrong results, because you would be able to select a different GPU.)